asab/library/providers/filesystem.py

```diff
@@ -31,6 +32,11 @@ def __init__(self, library, path, layer, *, set_ready=True):
 		'''
 
 		super().__init__(library, layer)
+		# Parse file:// URI if present
+		if path.startswith('file://'):
+			parsed_uri = urllib.parse.urlparse(path)
+			# Combine netloc and path to get the full path
+			path = os.path.join(parsed_uri.netloc, parsed_uri.path)
```

Comment:
You don't need this join ... netloc must be empty ... it is likely a result of using just two `/`